programs/Makefile

```diff
@@ -76,6 +78,10 @@ ifdef PTHREAD
 APPS +=	ssl/ssl_pthread_server$(EXEXT)
 endif
 
+ifdef TEST_CPP
+APPS += test/header_test$(EXEXT)
```

Comment:
Since this test tests C++ compatibility of the headers with full config. It is different from other tests/programs that all build with one valid config. Please check if a full config is valid for other programs.
If not I would rather make it mutually exclusive with all other programs.

Reply:
This tests headers with any config - there are usually no ifdef guards in headers, hence nothing really should change.

Reply:
In `all.sh` we should enable the full config and build this. The actual `Makefile` should just take the configs it's given.